modules/installation-configure-proxy.adoc

```diff
@@ -0,0 +1,76 @@
+// Module included in the following assemblies:
+//
+// * installing/installing_aws/installing-aws-customizations.adoc
+// * installing/installing_aws_user_infra/installing-aws-user-infra.adoc
+// * installing/installing_bare_metal/installing-bare-metal.adoc
+// * installing/installing_vsphere/installing-vsphere.adoc
+// TODO: gcp
+
+// Questions:
+// - Can they provide multiple certs in additionalTrustBundle?
+// - Is it true that the trustedCA field is filled in? Slides say "trusted-ca-bundle",  but ConfigMap created is user-ca-bundle
+
+[id="installation-configure-proxy_{context}"]
+= Configuring the cluster-wide proxy during installation
+
+Production environments can deny direct access to the Internet and instead have
+an HTTP or HTTPS proxy available. You can configure a new {product-title}
+cluster to use a proxy by configuring the proxy settings in the
+`install-config.yaml` file.
+
+.Prerequisites
+
+* An existing `install-config.yaml` file.
+
+.Procedure
+
+. Edit your `install-config.yaml` file and add the proxy settings. For example:
++
+[source,yaml]
+----
+apiVersion: v1
+baseDomain: my.domain.com
+proxy:
+  httpProxy: http://<username>:<pswd>@<ip>:<port> <1>
+  httpsProxy: https://<username>:<pswd>@<ip>:<port> <2>
+  noProxy: example.com <3>
+additionalTrustBundle: | <4>
+    -----BEGIN CERTIFICATE-----
+      <MY_TRUSTED_CA_CERT>
+    -----END CERTIFICATE-----
+...
+----
+<1> A proxy URL to use for creating HTTP connections outside the cluster. The
+URL scheme must be `http`.
```

Comment:
I see @danehans asking for this restriction [above][1], but I'm curious about why.  My understanding is that you can use a HTTPS URI for `httpProxy` (e.g. to avoid leaking your proxy-connection creds), although you might bump into containers/image#699 if the proxy doesn't support any of the encryption options currently compiled into containers/image (and that would go for `httpsProxy` too).

[1]: https://github.com/openshift/openshift-docs/pull/16635/files#r324255631

Reply:
I once had a try with using https URI for `httpProxy`, the network-operator will complain about it.

Reply:
What error message did you see?

Reply:
It's a quite long time ago, so I just spin up a new cluster with such setting. 
In network-operator pod log, I could see such message, also a clear prompt.
```
2019/09/27 02:42:21 Updated ClusterOperator with conditions:
- lastTransitionTime: "2019-09-27T02:42:21Z"
  message: The configuration is invalid for proxy 'cluster' (httpProxy requires a
    'http' URI scheme). Use 'oc edit proxy.config.openshift.io cluster' to fix.
  reason: InvalidProxyConfig
```

Reply:
Followed up [here][1], where that check landed.

[1]: https://github.com/openshift/cluster-network-operator/commit/095cd157b3a2f8764f7ee090d09a3a92cf177ef8#r35259609